test/RibbonTreasuryVaultWithAirSwap.ts

```diff
@@ -0,0 +1,3679 @@
+          const weth = assetContract.connect(userSigner);
+          await weth.deposit({ value: depositAmount });
+        }
+        await vault.connect(ownerSigner).commitAndClose();
+
+        await time.increaseTo((await getNextOptionReadyAt()) + DELAY_INCREMENT);
+
+        await vault.connect(keeperSigner).rollToNextOption();
+      });
+      it("completes the trade with the counterparty", async function () {
+        const otoken = await ethers.getContractAt(
+          "IERC20",
+          defaultOtoken.address
+        );
+
+        // Purchase all the oTokens
+        const initialOtokenBalance = await otoken.balanceOf(vault.address);
+        let decimals = premiumInStables ? premiumDecimals : tokenDecimals;
+        const bid = wmul(
+          initialOtokenBalance.mul(BigNumber.from(10).pow(10)),
+          firstOptionPremium
+        )
+          .div(BigNumber.from(10).pow(18 - decimals))
+          .toString();
+        // Assume premium is in stables
+        const signedOrder = await signOrderForAirSwap({
+          vaultAddress: vault.address,
+          counterpartyAddress: userSigner.address,
+          signerPrivateKey: userSignerPrivateKey,
+          sellToken: otoken.address,
+          buyToken: premiumContract.address,
+          sellAmount: initialOtokenBalance.toString(),
+          buyAmount: bid,
+        });
+
+        await premiumContract
+          .connect(userSigner)
+          .approve(airswapContract.address, bid);
+
+        const initialUserSignerBalance = await premiumContract
+          .connect(userSigner)
+          .balanceOf(userSigner.address);
+
+        // Before swapping, assert balances are 0 for accurate checking after
+        assert.bnEqual(
+          await premiumContract.connect(vault.address).balanceOf(vault.address),
+          BigNumber.from(0)
+        );
+        assert.bnEqual(
+          await otoken.connect(userSigner).balanceOf(userSigner.address),
+          BigNumber.from(0)
+        );
+
+        const res = await vault.connect(keeperSigner).sellOptions(signedOrder);
+        await expect(res).to.emit(airswapContract, "Swap");
+        await expect(res)
+          .to.emit(otoken, "Transfer")
+          .withArgs(vault.address, userSigner.address, initialOtokenBalance);
+        await expect(res)
+          .to.emit(premiumContract, "Transfer")
+          .withArgs(userSigner.address, vault.address, bid);
+
+        // Check that userSigner and vault tokens changed as expected
+        assert.bnEqual(
+          await otoken.balanceOf(vault.address),
+          BigNumber.from(0)
+        );
+
+        assert.bnEqual(
+          await premiumContract.connect(vault.address).balanceOf(vault.address),
+          BigNumber.from(bid)
+        );
+
+        assert.bnEqual(
+          await premiumContract
+            .connect(userSigner)
+            .balanceOf(userSigner.address),
+          BigNumber.from(initialUserSignerBalance).sub(bid)
+        );
+
+        assert.bnEqual(
```

Comment:
If checking whether a transfer has successfully occurred I would suggest using a testing pattern similar to:

const BalBefore = ...

const tx = (some tx occurs)

const BalAfter = ...

 assert.bnEqual(BalAfter.sub(BalBefore), amountTransferred)

Reply:
Fixed in [a30c9d0](https://github.com/ribbon-finance/ribbon-v2/pull/297/commits/a30c9d05c95b9d301e7c16500455108d96450e59)

Reply:
Addressed in [8928237](https://github.com/ribbon-finance/ribbon-v2/pull/297/commits/892823778c487e6773542058138ce27725910b5a)